From: Ben Hutchings <ben@decadent.org.uk>
Date: Fri, 21 Mar 2014 19:44:13 +0000
Subject: ARM: mm: Avoid ABI change in 3.13.6
Forwarded: not-needed

struct mem_type gained a new member in 3.13.6, and there are two
exported functions that take/return pointers to this structure.
However the actual definition is opaque to modules, so this is not an
ABI change and we can just hide it from genksyms.

--- a/arch/arm/mm/mm.h
+++ b/arch/arm/mm/mm.h
@@ -38,7 +38,10 @@ static inline pmd_t *pmd_off_k(unsigned
 
 struct mem_type {
 	pteval_t prot_pte;
+#ifndef __GENKSYMS__
+/* Added in 3.13.6, not an ABI change as the structure is opaque to modules */
 	pteval_t prot_pte_s2;
+#endif
 	pmdval_t prot_l1;
 	pmdval_t prot_sect;
 	unsigned int domain;
